### PR TITLE
6230 gf upgrade part deux

### LINKED ISSUE #6230: Upgrade from Glassfish 4.1
Dataverse is currently running on a (modified) Glassfish 4.1. This has served us well, but is old and no longer actively supported. Especially in light of the upcoming DataTags integration, we want to upgrade to a new app server for security uogrades, as well as being able to leverage newer features (e.g. Config API) and enable upgrades of other core technology (e.g. Java EE 7 -> Jakarta EE 8).

There are a few options to explore. Some work has already been done for Payara 5, and this tickets will serve for both  for the evaluation of that and some other candidates (TomEE, Glassfish 5, Wildfly), and the actual upgrade.

Note that I'm closing this old issue #2628 with a very similar title, as that is several years old now and was originally about upgrading to 4.1.1. 

Let's use this issue going forward for comments related to upgrading from glassfish 4.1.

(per request, also adding a link to the GH project that references subtasks related to this upgrade: https://github.com/orgs/IQSS/projects/16)

### PR DESCRIPTION
**Pull request by scolapasta**
**What this PR does / why we need it**:
Fixes issues on template and guestbook pages dealing with editMode

**Which issue(s) this PR closes**:
Closes #6230 

(though we may not want to close #6230 until #6595 testing is complete)

**Special notes for your reviewer**:
editMode viewparam changed to omnifaces for templates
editMode viewparam removed completely from guestbooks, as it was not used

**Suggestions on how to test this**:
Make sure you can add authors on templates; multiple choice questions / options on guestbook

**Does this PR introduce a user interface change?**:
No

**Is there a release notes update needed for this change?**:
No

**Additional documentation**:
this code change was made to work with Payara5, but is backwards compatible with Glassfish 4.1

### CODE REVIEW

File: src/main/webapp/guestbook.xhtml [hunk 1]
```diff
@@ -197,4 +196,4 @@
             </ui:define>
         </ui:composition>
     </h:body>
-</html>
\ No newline at end of file
+</html>
```
Comment (mheppler) [inline src/main/webapp/guestbook.xhtml:199]: Weird. Couldn't tell what, if anything, changed on this line. Same with the template.xhtml file.

File: src/main/webapp/template.xhtml [hunk 2]
```diff
@@ -19,7 +20,7 @@
                 <f:metadata>
                     <f:viewParam name="id" value="#{TemplatePage.templateId}"/>
                     <f:viewParam name="ownerId" value="#{TemplatePage.ownerId}"/>
-                    <f:viewParam name="editMode" value="#{TemplatePage.editMode}"/>
+                    <o:viewParam name="editMode" value="#{TemplatePage.editMode}"/>
```
Comment (mheppler) [inline src/main/webapp/template.xhtml:23]: Why was this editMode viewParam removed on guestbook.xhtml, but converted to an OmniFaces component on this file? Does not seem to be utilized on that page, as far as I can tell.

File: src/main/webapp/template.xhtml [hunk 3]
```diff
@@ -127,4 +128,4 @@
             </ui:define>
         </ui:composition>
     </h:body>
-</html>
\ No newline at end of file
+</html>
```
Comment (mheppler) [inline src/main/webapp/template.xhtml:131]: Same comment as on guestbook.xhtml. What changed?

File: src/main/webapp/template.xhtml (hunk 2 above)
Comment (sekmiller) [reply src/main/webapp/template.xhtml:23]: It's used in manageTemplates where you can edit a template and the edit button passes the edit mode in the url:

<h:outputLink value="/template.xhtml?id=#{template.id}&#38;ownerId=#{manageTemplatesPage.dataverse.id}&#38;editMode=METADATA">
                                                    <h:outputText value="#{bundle['dataset.manageTemplates.tab.action.btn.edit.metadata']}" />
                                                </h:outputLink>

**Comment by kcondon** [comment]
@scolapasta 

i'm seeing this in logs when I add multiple options
5:45
[2020-04-07T17:44:39.554-0400] [glassfish 4.1] [WARNING] [] [javax.enterprise.resource.webcontainer.jsf.renderkit] [tid: _ThreadID=55 _ThreadName=jk-connector(2)] [timeMillis: 1586295879554] [levelValue: 900] [[
Unable to find component with ID questionResponseOptions in view.]]
5:48
happens every time you click the + button on response options to multiple choice questions.

**Comment by scolapasta** [comment]
The same thing occurs with dev branch, that is this is not introduced by this change. Since this PR isn't the cause and this allows us to get past some issues on payara5, I'd like to still merge this now.

**Pull request merged**